components/Releases/CreateRelease.jsx

```diff
@@ -0,0 +1,176 @@
+import { useState, useEffect } from "react"
+import { useUser, useSupabaseClient } from "@supabase/auth-helpers-react"
+
+const releaseTypes = [
+  { id: 1, text: "LP" },
+  { id: 2, text: "EP" },
+  { id: 3, text: "Single" },
+  { id: 4, text: "Compilation" },
+  { id: 5, text: "Soundtrack" },
+  { id: 6, text: "Choose release type", isDisabled: true },
+]
+
+export default function CreateRelease({ user, setCreateNewRelease }) {
+  const supabase = useSupabaseClient()
+  const [title, setTitle] = useState()
+  const [artist, setArtist] = useState(
+    user.user_metadata.type === "Artist" ? user.user_metadata.name : null
+  )
+  const [label, setLabel] = useState(
+    user.user_metadata.type === "Label" ? user.user_metadata.name : null
+  )
+  const [artwork_url, setArtwork_url] = useState()
+  const [download_url, setDownload_url] = useState()
+  const [page_password, setPage_password] = useState()
+  const [is_password_protected, setIs_password_protected] = useState(false)
+  const [type, setType] = useState(releaseTypes[5].text)
+
+  const [user_id, setUser_id] = useState(user.id)
+
+  async function createNewRelease({
+    title,
+    artist,
+    label,
+    artwork_url,
+    download_url,
+    type,
+    user_id,
+  }) {
+    try {
+      let slugger = title.toLowerCase()
+      let newRelease = {
+        title: title,
+        artist: artist,
+        label: label,
+        artwork_url: artwork_url,
+        download_url: download_url,
+        type: type,
+        release_slug: slugger
+          .replace(/[^a-z0-9 -]/g, "")
+          .replace(/\s+/g, "-")
+          .replace(/-+/g, "-"),
+        user_id: user_id,
+      }
+      const { data, error } = await supabase
+        .from("releases")
+        .insert([newRelease])
+      if (error) throw error
+      alert("New release created!")
+    } catch (error) {
+      alert("Error creating new release!")
+    } finally {
+      setCreateNewRelease(false)
```

Comment:
Two things here:
* Where is this function defined?
* The naming should just be `createNewRelease()` if that's what it is intended to do.